staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -315,23 +320,99 @@ func strategicPatchObject(
 
 type applyPatcher struct {
 	*patcher
+
+	model proto.Schema
 }
 
-func (p *applyPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (runtime.Object, error) {
-	versionedObjToUpdate, err := p.creater.New(p.kind)
+// TODO(apelisse): workflowId needs to be passed as a query
+// param/header, and a better defaulting needs to be defined too.
+const workflowId = "default"
+
+func (p *applyPatcher) convertCurrent(obj runtime.Object) (map[string]interface{}, error) {
```

Comment:
These names are an improvement, thanks--

I am a little uneasy with names that have the word "convert" in them but don't say what they are going to convert to or from.

I also think tracking the versions is going to be critical and maybe should be a first-class concept here (e.g., tracked in parameters or some such). Arguably that can happen later, but then I'd still like these names to be like convertCurrentToMergeVersion or currentToExternal, or something.